core/domain/feedback_services.py

```diff
@@ -235,21 +296,54 @@ def update_messages_read_by_the_user(user_id, thread_id, message_ids):
     feedback_thread_user_model.put()
 
 
-def add_message_id_to_read_by_list(thread_id, user_id, message_id):
-    """Adds the message id to the list of message ids read by the user.
+def add_message_ids_to_read_by_list(thread_ids, user_id, message_ids):
+    """Adds the message ids to the list of message ids read by the user.
 
     Args:
-        thread_id: str. The id of the thread.
+        thread_ids: list(str). The ids of the threads.
         user_id: str. The id of the user reading the messages.
-        message_id: int. The id of the message.
-    """
-    feedback_thread_user_model = (
-        feedback_models.GeneralFeedbackThreadUserModel.get(
-            user_id, thread_id) or
-        feedback_models.GeneralFeedbackThreadUserModel.create(
-            user_id, thread_id))
-    feedback_thread_user_model.message_ids_read_by_user.append(message_id)
-    feedback_thread_user_model.put()
+        message_ids: list(str). The ids of the messages.
+    """
+    # First get all of the GeneralFeedbackThreadUserModels that already exist.
+    current_feedback_thread_user_models = (
+        feedback_models.GeneralFeedbackThreadUserModel.get_multi(
+            user_id, thread_ids))
+
+    # Keep track of which thread_ids do not have models yet.
+    missing_thread_ids = []
+    # Keep track of the current_feedback_thread_user_models that aren't None.
+    feedback_thread_user_models = []
+    for index, feedback_model in enumerate(
```

Comment:
I don't think feedback_model is appropriate here -- call it feedback_thread_user_model.